clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -307,11 +307,22 @@ public void completed(
         // Use thread pool to handle the completion reporting to make sure it is not blocking the report.
         if (isTopicPartitionIngesting(kafkaTopic, partition)) {
           getCompletionHandlingExecutor().submit(() -> {
-            VeniceStoreVersionConfig config = getConfigLoader().getStoreConfig(kafkaTopic);
-            config.setRestoreDataPartitions(false);
-            config.setRestoreMetadataPartition(false);
-            // Start partition consumption locally.
-            startConsumption(config, partition, leaderState);
+            /**
+             * Start partition consumption locally.
+             * If any error happens when starting the consumption, error will be reported.
+             */
+            try {
+              VeniceStoreVersionConfig config = getConfigLoader().getStoreConfig(kafkaTopic);
+              config.setRestoreDataPartitions(false);
+              config.setRestoreMetadataPartition(false);
+              startConsumption(config, partition, leaderState);
+            } catch (Exception e) {
```

Comment:
This logic is a must!
During the handover from II process to main process, `startConsumption` could fail since the database reopen could breach the memory limit.

Reply:
I see...oh right got it. I mean before that, I never see this part failed, but this new logic do have the chance to fail.